examples/opensora_hpcai/opensora/models/stdit/stdit2.py

```diff
@@ -385,7 +389,7 @@ def construct(
         T, H, W = self.get_dynamic_size(x)
         S = H * W
         scale = rs / self.input_sq_size
-        base_size = round(S**0.5)
+        base_size = int(round(S ** Tensor(0.5)))
```

Comment:
`int(Tensor)`?

Reply:
> `int(Tensor)`?

int(round(x)) and int() does not behave same

Reply:
1. `round` already returns `int`. 2. Why need to convert tensor back to int?

Reply:
it is a bit trick,
In graph mode, `round` will be interpreted as `ops.Round` first and try to read a tensor. If it fails (like reading a int) , then it fall backs to the python interpreter silently, and run `round(x)` once and make it as a constant. So to avoid fallback, the input of round should be a Tensor first, that is why added a Tensor(0.5) first instead of 0.5, and the output of round is then a float tensor.

Then, `int` will be interpreted as `ops.Cast`, and converted the float tensor to int tensor. So finally `base_size` is an integer tensor.

P.S., to know what is happening in side the graph, may need to read the IR then ...

Reply:
I see. Well, `base_size` doesn't need to be an `int`

Reply:
yeah then it should be safe to remove cast